query-node/mappings/workingGroups.ts

```diff
@@ -800,9 +801,15 @@ export async function workingGroups_WorkerExited({ store, event }: EventContext
   })
 
   await store.save<WorkerExitedEvent>(workerExitedEvent)
-  ;(worker.status as WorkerStatusLeft).workerExitedEventId = workerExitedEvent.id
+
+  const newStatus = new WorkerStatusLeft()
+  newStatus.workerStartedLeavingEventId = (worker.status as WorkerStatusLeaving).workerStartedLeavingEventId
+  newStatus.workerExitedEventId = workerExitedEvent.id
+
+  worker.status = newStatus
   worker.stake = new BN(0)
   worker.rewardPerBlock = new BN(0)
+  worker.missingRewardAmount = undefined
```

Comment:
Just to be sure – this means that, if the worker left the owed reward is disregarded?

Reply:
The scenario I'm thinking of is:

1.  A worker has missed reward due to an insufficient WG budget.
2.  After the unstaking period there's still: a) missing reward and b) insufficient budget

I think that there will not be payment if the worker is not in the working group, so I think that I have my answer :)

Reply:
Judging by the code in https://github.com/Joystream/joystream/blob/olympia/runtime-modules/working-group/src/lib.rs#L1262, the runtime will try to pay the missed reward just before removing the worker. Once the worker is removed, the missed reward is "gone" (since the worker is no loger part of the `WorkerById` map so the runtime effectively doesn't recognize it anymore), therefore it should be no longer part of the debt.